core/src/global/ionic-global.ts

```diff
@@ -227,7 +227,7 @@ export const initialize = (userConfig: IonicConfig = {}) => {
      * behavior for applications that are not setting the "theme".
      */
     while (elm) {
-      const theme = (elm as any).theme || elm.getAttribute('theme');
```

Comment:
Would this issue go away if we made the `mode` a virtual prop on router outlet instead?

Reply:
works for me 👍